csrc/scheduler/greedy.cpp

```diff
@@ -236,10 +236,9 @@ class CompileTimeChecker : private IterVisitor {
     auto inp = scatter->in()->as<TensorView>();
     auto out = scatter->out()->as<TensorView>();
 
-    if (out->getLogicalDomain().size() != 1) {
+    if (!scatter->exactSizes()) {
```

Comment:
nitpick: conceptually we can still handle 1d input, because exactSizes() flag is non consequential for 1d inputs.
Not that important since we are also hiding the flag behind c++ api.